deploy/charts/approver-policy/templates/deployment.yaml

```diff
@@ -38,8 +38,12 @@ spec:
         image: "{{ template "image" (tuple .Values.image $.Chart.AppVersion) }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         ports:
-        - containerPort: {{ .Values.app.webhook.port }}
-        - containerPort: {{ .Values.app.metrics.port }}
+        - name: https
+          containerPort: {{ .Values.app.webhook.port }}
+        - name: http-metrics
+          containerPort: {{ .Values.app.metrics.port }}
+        - name: healthcheck
+          containerPort: {{ .Values.app.readinessProbe.port }}
```

Comment:
```suggestion
```

Why do we need to add a named port for the health checks? I don't see the benefit.

Reply:
> Why do we need to add a named port for the health checks? I don't see the benefit.

This is so that network policies for health checks can target the functional name, rather than the port number. [Here is the netpol that I've deployed for approver-policy](https://github.com/solidDoWant/infra-mk3/blob/master/cluster/gitops/certificates/approver-policy/app/netpol.yaml#L29). Note that while this is a cilium-specific resource, the same concept applies to the general k8s networkpolicy resource as well. Whether or not probes from the host/kubelet are allowed by default is CNI-dependent.

Reply:
I see, in all the clusters I work on it must be enabled by default. 😉 Then it makes sense! But could we please use functional port names and not mix in protocols etc.?

Reply:
Yea from what I've gathered, most CNIs have specific logic to allow for this. However they do see the traffic:

![image](https://github.com/user-attachments/assets/97f8f214-566f-4b1c-a361-85fc7335aab7)

And under certain conditions require explicit policies the probe port(s). [Here's an example with](https://github.com/istio/istio/issues/49277) Cilium + Istio in eBPF mode where probes fail because Cilium expects them to come from the host, but they instead come from a link-local address.